### ISSUE TITLE
Support `format: uuid`

### ISSUE THREAD

**Issue by liambutler-lawrence** [apple/swift-openapi-generator#707]
### Motivation

Our API spec heavily uses UUID-typed fields, which are represented in our OpenAPI doc using `format: uuid` as described in the [JSONSchema spec](https://json-schema.org/understanding-json-schema/reference/string#resource-identifiers).

However, `swift-openapi-generator` currently ignores that part of the OpenAPI doc and generates them all as `Swift.String`, meaning we have to manually call `UUID.uuidString` and `UUID(uuidString:)` (and implement required error handling on the latter).

### Proposed solution

Make the generator respect `format: uuid` and define such fields using the Swift `UUID` type.

**Comment by simonjbeaumont** [comment]
Thanks for filing this @liambutler-lawrence.

This is a duplicate of #375, which is a more general ask for generating more typed values.

There's an open PR, #627, that specifically adds support for UUID, but that's currently blocked on working out a backwards compatibility story (see https://github.com/apple/swift-openapi-generator/pull/627#issuecomment-2518138710).
